src/main/ArrayManager.java

```diff
@@ -160,11 +181,11 @@ public void shuffleArray(int[] array, int currentLen, ArrayVisualizer ArrayVisua
             Delays.setSleepRatio(sleepRatio);
         }
 
-        Shuffles tempShuffle = this.Shuffles;
-        if(Distributions == Distributions.RANDOM || Distributions == Distributions.EQUAL)
-            this.Shuffles = Shuffles.ALREADY;
-        Shuffles.shuffleArray(array, this.ArrayVisualizer, Delays, Highlights, Writes);
-        this.Shuffles = tempShuffle;
+        ShuffleGraph tempShuffle = this.shuffle;
+        if(distribution == Distributions.RANDOM || distribution == Distributions.EQUAL)
+            this.shuffle = ShuffleGraph.single(Shuffles.ALREADY);
```

Comment:
this doesn't make sense anymore, since you can change element values

Reply:
True